sql/wsrep_applier.cc

```diff
@@ -78,6 +78,20 @@ Format_description_log_event *wsrep_get_apply_format(THD *thd) {
   return thd->wsrep_rli->get_rli_description_event();
 }
 
+static uint wsrep_errno_for_voting(uint err) {
+  static std::map<uint, uint> errno_map = {
+      {ER_NO_REFERENCED_ROW_2, ER_NO_REFERENCED_ROW},
+      {ER_ROW_IS_REFERENCED_2, ER_ROW_IS_REFERENCED}};
+
+  if (errno_map.count(err)) {
+    auto res = errno_map[err];
```

Comment:
I am generally scared of this usage map[elem]. I have seen subtle bugs. Your usage is fine because you verify that the element exists.

If the element doesn't exist, errno_map[err] will insert a new element to the map.

Any new code might not be aware that count() should be used first. Please use find with iterator.

Reply:
i asked this question as github comment, may be better to ask here. The commit you mentioned is introduced in Server on on Jun 19, 2018. Why are we seeing this now? :thinking: